Scripts/assignment/assignment_period.py

```diff
@@ -142,6 +142,10 @@ def assign(self, matrices, iteration):
                 path_not_found = mtxs["time"][mtx_class] > 999999
                 mtxs[mtx_type][mtx_class][path_not_found] = 999999
         # adjust impedance
+        if iteration == "last":
+            for mtx_class in ("trailer_truck", "truck"):
+                path_not_found = mtxs["time"][mtx_class] <= 999999
+                mtxs["cost"][mtx_class][path_not_found] = 0
```

Comment:
Why not simply like this? I think this will give the same result.

```suggestion
                mtxs["cost"][mtx_class] = 0
```

Reply:
Because before, there was 999999s in the cost matrices despite the comment in L139: https://github.com/HSLdevcom/helmet-model-system/blob/6b3d1ebd1ead1f464ef8fecdc5f4a13c460433c7/Scripts/assignment/assignment_period.py#L139-L143 So I mimiced the behavior.

Reply:
I see.

The comment in L139 is actually a bit confusing, as it describes the problem and not the solution. The solution is that dist and cost is set to 999999 if no path is found.

In your code it is actually the bool matrix name `path_not_found` that is confusing. According to your definition, it represents the opposite (i.e., `path_found`).

Reply:
I *think* a more elegant solution could be to initialize the impedance matrices to 0 here:

https://github.com/HSLdevcom/helmet-model-system/blob/6b3d1ebd1ead1f464ef8fecdc5f4a13c460433c7/Scripts/assignment/emme_assignment.py#L86-L91

But I am not 100 % sure that it does not have any side effects.

Reply:
Ooops, you are totally right on the naming, I though I fixed it but I did not. 🙈 I made a new commit to fix that and docs.

Changing the default looks risky!